test/SILGen/deinit_recursive.swift

```diff
@@ -0,0 +1,91 @@
+// RUN: %target-swift-emit-silgen %s | %FileCheck %s
+
+class Node {
```

Comment:
Can you add a test for generic classes? One example should have the recursive link node use the same instantiation of the arguments, eg:

```
class Node<A, B> { var next: Node<A, B> }
```

another one should test a different instantiation:

```
class Node<A, B> { var next: Node<Int, (A, B)>? }
```

Presumably only the former should be optimized!